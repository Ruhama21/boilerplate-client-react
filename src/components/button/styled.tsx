import styled from 'styled-components';
import { Button } from 'react-bootstrap';

export const CustomButton = styled(Button)`
  background-color: ${(props) => props.theme.buttons.backgroundColor};
  border: none;
  color: ${(props) => props.theme.buttons.text};
  /* color: ${(props) => props.theme.color}; */
  width: ${(props) => props.theme.width};
  */ &:hover {
    background-color: ${(props) => props.theme.buttons.hoverBackgroundColor};
    /* border-color: ${(props) => props.theme.buttons.hoverBorderColor}; */
  }
  &:disabled {
    background-color: ${(props) => props.theme.buttons.disabledBackgroundColor};
    /* border-color: ${(props) => props.theme.disabledBorderColor}; */
  }
`;

// CustomButton.defaultProps = {
//   theme: {
//     backgroundColor: accent,
//     borderColor: accent,
//     color: '#fff',
//     disabledBackgroundColor: accentDisabled,
//     disabledBorderColor: accent,
//     hoverBackgroundColor: accent,
//     hoverBorderColor: accent,
//     width: '90%',
//   },
// };
