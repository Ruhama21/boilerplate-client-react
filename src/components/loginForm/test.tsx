import { act, render, screen } from '@testing-library/react'
import userEvent from '@testing-library/user-event'
import { LoginForm } from './';
import { errorMessages } from './schema';

describe('LoginForm', () => {
  const validEmail = 'test@email.com'
  const validPassword = 'Password123!'

  let emailField: HTMLElement
  let passwordField: HTMLElement
  let submitButton: HTMLElement
  const mockOnSubmit = jest.fn()

  beforeEach(() => {
    render(<LoginForm onSubmit={mockOnSubmit} />)
    emailField = screen.getByLabelText(/email/i)
    passwordField = screen.getByLabelText(/password/i)
    submitButton = screen.getByRole('button', {
      name: /submit/i,
    })
    mockOnSubmit.mockReset()
  })

  it('should render', () => {
    expect(emailField).toBeInTheDocument()
    expect(passwordField).toBeInTheDocument()
    expect(submitButton).toBeInTheDocument()
  })

  describe('valid input', () => {
    it('should call onSubmit once formData object including username and password', async () => {
      userEvent.type(emailField, validEmail)
      userEvent.type(passwordField, validPassword)
      await act(async () => {
        userEvent.click(submitButton)
      })

      const formData = {
        email: validEmail, 
        password: validPassword
      }

      expect(mockOnSubmit).toHaveBeenCalledTimes(1)
      expect(mockOnSubmit).toHaveBeenCalledWith(formData)
    })

    it('should not display error messages', async () => {
      userEvent.type(emailField, validEmail)
      userEvent.type(passwordField, validPassword)
      await act(async () => {
        userEvent.click(submitButton)
      })

      expect(screen.queryAllByRole('alert')).toHaveLength(0)
    })
  })

  describe('invalid input', () => {
    it('should not call onSubmit', async () => {
      userEvent.type(emailField, '')
      userEvent.type(passwordField, '')
      await act(async () => {
        userEvent.click(submitButton)
      })

      expect(mockOnSubmit).not.toHaveBeenCalled()
    })

    it('should display error messages', async () => {
      userEvent.type(emailField, 't')
      userEvent.type(passwordField, 't')
      await act(async () => {
        userEvent.click(submitButton)
      })

      expect(screen.getAllByRole('alert')).toHaveLength(2)
    })

    describe('invalid email', () => {
      it('should only display email error message', async () => {
        userEvent.type(emailField, '')
        userEvent.type(passwordField, validPassword)
        await act(async () => {
          userEvent.click(submitButton)
        })

        expect(screen.getAllByRole('alert')).toHaveLength(1)
        expect(screen.getByRole('alert').innerHTML).toBe(errorMessages.EMAIL_REQUIRED)
      })
    })

    describe('invalid password', () => {
      it('should only display password error message', async () => {
        userEvent.type(emailField, validEmail)
        userEvent.type(passwordField, '')
        await act(async () => {
          userEvent.click(submitButton)
        })

        expect(screen.getAllByRole('alert')).toHaveLength(1)
        expect(screen.getByRole('alert').innerHTML).toBe(errorMessages.PASSWORD_REQUIRED)
      })
    })
  })
})