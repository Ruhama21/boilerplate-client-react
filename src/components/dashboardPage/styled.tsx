import styled from 'styled-components';

export const DashboardPageContainer = styled.div`
    backgroundColor: #000;
`;