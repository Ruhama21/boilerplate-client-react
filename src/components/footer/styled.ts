import styled from 'styled-components'

export const FooterWrapper = styled.div`
  width: 100%;
  text-align: center;
  border-top: 2px solid black;
`
