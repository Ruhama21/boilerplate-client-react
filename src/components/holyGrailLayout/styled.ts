import styled from 'styled-components';
import {} from '../../utils/colors';

export const HolyGrailWrapper = styled.div`
  display: grid;
  height: 100vh;
  grid-template-columns: 1fr 2fr 1fr;
  grid-template-rows: minmax(200px, auto);
  grid-gap: 1em;
  @media all and (min-width: 768px) {
    flex-direction: row;
    flex-wrap: wrap;
  }
`;

export const HolyGrailNavLeft = styled.nav`
  @media all and (min-width: 768px) {
    order: 1;
    flex: 1;
  }
`;

export const HolyGrailMain = styled.main`
  display: flex;
  flex: 1;
  justify-content: center;
  align-items: center;
  @media all and (min-width: 768px) {
    flex: 2;
    order: 2;
    min-height: 80vh;
  }
`;

export const HolyGrailFooter = styled.footer`
  grid-column: 1 / -1;
  width: 100%;
  order: 3;
`;
