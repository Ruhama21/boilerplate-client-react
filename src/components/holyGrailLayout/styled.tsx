import styled from 'styled-components';
import {} from '../../utils/styleValues';

export const HolyGrailWrapper = styled.div`
  display: flex;
  flex-direction: column;
`;

export const HolyGrailMainWrapper = styled.div`
  display: flex;
  flex: 1;
  flex-direction: column;
  max-height: calc(100vh - ${(props) => props.theme.footerHeight});

  @media (min-width: 768px) {
    flex-direction: row;
    flex: 1;
  }
`;

export const HolyGrailLeftAside = styled.aside`
  order: -1;
  flex: 0 0 12em;
  @media (min-width: 768px) {
    flex: 0 0 12em;
  }
`;

export const HolyGrailMain = styled.main`
  flex: 1;
  @media (min-width: 768px) {
    flex: 1;
  }
`;

export const HolyGrailRightAside = styled.aside`
  flex: 0 0 12em;

  @media (min-width: 768px) {
    flex: 0 0 12em;
  }
`;
