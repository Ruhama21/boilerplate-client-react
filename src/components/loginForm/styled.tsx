import Button from 'react-bootstrap/Button';
import Form from 'react-bootstrap/Form';
import styled from 'styled-components';

export const StyledForm = styled(Form)`
  display: flex;
  justify-content: space-between;
  align-items: center;
  /* margin: 20px; */
  border: 2px solid red;
  padding: 20px;
`;

export const Login = styled.div``;

export const Register = styled.div``;

export const Title = styled.div`
  color: black;
  font-size: 2em;
  font-style: bold;
  padding-bottom: 10px;
`;

export const FormLabel = styled(Form.Label)`
  color: white;
`;

export const InputError = styled.span`
  color: lightgrey;
`;

export const ButtonWrapper = styled.div`
  display: flex;
  justify-content: center;
  align-items: center;
  padding-top: 10px;
`;

export const CancelButton = styled(Button)`
  color: white;
  background-color: ${(props) => props.theme.primary};
  border: 1px solid #679daa;
  margin-right: 20px;
`;

export const LogInButton = styled(Button)`
  color: white;
  background-color: ${(props) => props.theme.accent};
  border: none;
`;

export const CreateAccountButton = styled(Button)``;

export const Text = styled.div`
  color: white;
`;

export const LinkWrapper = styled.div`
  padding-bottom: 30px;
  font-size: 1.2em;
`;
