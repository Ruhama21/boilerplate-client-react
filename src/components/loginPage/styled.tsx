import styled from 'styled-components';

export const LoginFormContainer = styled.div`
  display: flex;
  flex-direction: colunn;
  border: 2px solid black;
  padding: 200px;
`;
