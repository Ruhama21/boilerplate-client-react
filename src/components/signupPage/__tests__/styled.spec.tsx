import { SignupPageContainer, SignupFormContainer } from '../styled';
import { snapshotMatch } from '../../../utils/test';
import 'jest-styled-components';

describe('Reset Password Page Styled Components', () => {
  it('Should match the stored <SignupPageContainer /> snapshot', () => snapshotMatch(<SignupPageContainer />));
  it('Should match the stored <SignupFormContainer /> snapshot', () => snapshotMatch(<SignupFormContainer />));
});