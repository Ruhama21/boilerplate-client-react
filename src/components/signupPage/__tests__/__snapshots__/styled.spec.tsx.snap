// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Reset Password Page Styled Components Should match the stored <SignupFormContainer /> snapshot 1`] = `
<div
  className=""
/>
`;

exports[`Reset Password Page Styled Components Should match the stored <SignupPageContainer /> snapshot 1`] = `
<div
  className=""
/>
`;
